openedx/features/partners/forms.py

```diff
@@ -114,8 +114,8 @@ def clean_registration_data(self, registration_data):
         registration_data.update(self.cleaned_data)
 
     def clean_country(self):
-        country = self.cleaned_data['country']
-        cleaned_country = next((code for code, label in COUNTRIES.items() if label.lower() == country.lower()), None)
+        country_code = self.cleaned_data['country']
+        cleaned_country = next((code for code in COUNTRIES.keys() if code == country_code), None)
```

Comment:
I guess we are keeping this just for validation if key exists or not let's please use `.get` for this purpose.